# This file should contain all the record creation needed to seed the database with its default values.
# The data can then be loaded with the rake db:seed (or created alongside the db with db:setup).
#
# Examples:
#
#   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
#   Mayor.create(name: 'Emanuel', city: cities.first)

organism = Organism.create(name: 'yeast', latin: 'Saccharomyces cerevisiae', gbtaxonid: 4932)
feature = Feature.create(so: 'SO:0000167', name: 'promoter', namespace: 'sequence', definition: 'A regulatory_region composed of the TSS(s) and binding sites for TF_complexes of the basal transcription machinery.')
sequence = Sequence.create(data: 'AATATTTGTATTGTTTTATGTAATTATAATCTTAAGCTTGCAATACCAATCACTATTACCATGTTAATGATCCCCCAGATAAAAAGACAAGACAGAGCATATATCAGACTACATTAACATCTGTACACCCCCAACTGCTTTGAAAAAACCCATGCAGTTATTATTTTCAACAGACCAAGTTTTTTTATCGCTTATCCGGCTCACAGAACGAGCCGTACAACCGTCATCAGACTTGCATGGAAATGTTGAAAAAGAGGTAAAAAAGAAAATATCAAACCCAAACACGCTCGGGCTGAAAAATTAATATGGCAGGAAACGAACTGTAGCCATAATGTACCTGTTCCCGCTAATAGTAATACCATCGTATTGCTAAGGTTAGTAGCTAAAAATCTCTTACTTTCTTTATTTTGACTCTTAGGTCTTACAAGCAATACAAAACCAACACACCTATATATAACTAATA')
location = Location.create(organism: organism, chromosome: 'chr02', start: 60736, stop: 61198, strand: -1 )
part = Part.create(name: 'yeast_promoter_YBL087C', sequence: sequence, feature:feature, location: location)
user = User.create(fullname: 'Giovanni Stracquadanio', provider: 'facebook', email: 'stracquadanio@jhu.edu')
user = User.create(fullname: 'Nicholas Culbertson', provider: 'google', email: 'nculbertson1@gmail.com')
vendor = Vendor.create(name: 'IDT')
oligoPlate = OligoPlate.create(name: 'JHU_PARTS_2012_OLIGO_1', vendor: vendor, user: user)
oligo = Oligo.create(part: part, name: 'yeast_promoter_YBL087C.o01', start: 1, stop: 150, sequence: 'AATATTTGTATTGTTTTATGTAATTATAATCTTAAGCTTGCAATACCAATCACTATTAC')
well = OligoPlateWell.create(oligo: oligo, oligo_plate: oligoPlate, well: 'A01')
plasmid = Plasmid.create(vendor: vendor, name: "pGEM-T")

device_type = DeviceType.create(value: "pcr")
device = Device.create(device_type_id: 1, name: "Test PCR Machine")
status1 = Status.create(process: "Spcr", name: "Pending", default: 1 )
status2 = Status.create(process: "Spcr", name: "Finished")
status3 = Status.create(process: "Tpcr", name: "Pending", default: 1 )
status4 = Status.create(process: "Tpcr", name: "Finished")
status5 = Status.create(process: "Fpcr", name: "Pending", default: 1 )
status6 = Status.create(process: "Fpcr", name: "Finished")
status7 = Status.create(process: "PcrGel", name: "Pending", default: 1 )
status8 = Status.create(process: "PcrGel", name: "Finished")
status9 = Status.create(process: "Ligation", name: "Pending", default: 1 )
status10 = Status.create(process: "Ligation", name: "Finished")
status11 = Status.create(process: "Transformation", name: "Pending", default: 1 )
status12 = Status.create(process: "Transformation", name: "Finished")
status13 = Status.create(process: "Clone", name: "Pending", default: 1 )
status14 = Status.create(process: "Clone", name: "Finished")
status15 = Status.create(process: "Cspcr", name: "Pending", default: 1 )
status16 = Status.create(process: "Cspcr", name: "Finished")
status17 = Status.create(process: "CspcrGel", name: "Pending", default: 1 )
status18 = Status.create(process: "CspcrGel", name: "Finished")
status19 = Status.create(process: "Sequencing", name: "Pending", default: 1 )
status20 = Status.create(process: "Sequencing", name: "Finished")
status21 = Status.create(process: "Sequencing", name: "Mailed")

qc1 = QualityControl.create(process: "Spcr", name: "PASS")
qc2 = QualityControl.create(process: "Spcr", name: "FAIL")
qc3 = QualityControl.create(process: "Tpcr", name: "PASS")
qc4 = QualityControl.create(process: "Tpcr", name: "FAIL")
qc5 = QualityControl.create(process: "Fpcr", name: "PASS")
qc6 = QualityControl.create(process: "Fpcr", name: "FAIL")
qc7 = QualityControl.create(process: "PcrGelLane", name: "PASS")
qc8 = QualityControl.create(process: "PcrGelLane", name: "FAIL")
qc9 = QualityControl.create(process: "Ligation", name: "PASS")
qc10 = QualityControl.create(process: "Ligation", name: "FAIL")
qc11 = QualityControl.create(process: "Transformation", name: "PASS")
qc12 = QualityControl.create(process: "Transformation", name: "FAIL")
qc13 = QualityControl.create(process: "Cloning", name: "PASS")
qc14 = QualityControl.create(process: "Cloning", name: "FAIL")
qc15 = QualityControl.create(process: "Cspcr", name: "PASS")
qc16 = QualityControl.create(process: "Cspcr", name: "FAIL")
qc17 = QualityControl.create(process: "CspcrGel", name: "PASS")
qc18 = QualityControl.create(process: "CspcrGel", name: "FAIL")
qc19 = QualityControl.create(process: "Sequencing", name: "PASS")
qc20 = QualityControl.create(process: "Sequencing", name: "FAIL")
qc20 = QualityControl.create(process: "Sequencing", name: "CHECK")
qc20 = QualityControl.create(process: "Sequencing", name: "FIXABLE")
qc20 = QualityControl.create(process: "Sequencing", name: "MISLABELLED")

prot_reag1 = ProtocolReagent.create(protocol_id: "6", name: "H2O", stock_concentration: "-", final_concentration: "-", volume_reaction: "14.25 ul")
prot_reag2 = ProtocolReagent.create(protocol_id: "6", name: "Herculase Bffer", stock_concentration: "5 x", final_concentration: "1 x", volume_reaction: "5.0 ul")
prot_reag3 = ProtocolReagent.create(protocol_id: "6", name: "dNTPs", stock_concentration: "2.5 mM", final_concentration: "250 uM", volume_reaction: "2.5 ul")
prot_reag4 = ProtocolReagent.create(protocol_id: "6", name: "Genomic DNA Template", stock_concentration: "-", final_concentration: "-", volume_reaction: "0.5 ul")
prot_reag5 = ProtocolReagent.create(protocol_id: "6", name: "1:5 Diluted Herculase DNA Pol", stock_concentration: "-", final_concentration: "-", volume_reaction: "0.75 ul")
prot_reag6 = ProtocolReagent.create(protocol_id: "6", name: "Total", stock_concentration: "-", final_concentration: "-", volume_reaction: "23.0 ul")
prot_reag7 = ProtocolReagent.create(protocol_id: "1", name: "H2O", stock_concentration: "-", final_concentration: "-", volume_reaction: "14.25 ul")
prot_reag8 = ProtocolReagent.create(protocol_id: "1", name: "Herculase Bffer", stock_concentration: "5 x", final_concentration: "1 x", volume_reaction: "5.0 ul")
prot_reag9 = ProtocolReagent.create(protocol_id: "1", name: "dNTPs", stock_concentration: "2.5 mM", final_concentration: "250 uM", volume_reaction: "2.5 ul")
prot_reag10 = ProtocolReagent.create(protocol_id: "1", name: "1:5 Diluted Herculase DNA Pol", stock_concentration: "-", final_concentration: "-", volume_reaction: "0.75 ul")
prot_reag11 = ProtocolReagent.create(protocol_id: "1", name: "Total", stock_concentration: "-", final_concentration: "-", volume_reaction: "22.5 ul")
prot_reag12 = ProtocolReagent.create(protocol_id: "2", name: "H2O", stock_concentration: "-", final_concentration: "-", volume_reaction: "14.75 ul")
prot_reag13 = ProtocolReagent.create(protocol_id: "2", name: "Herculase Bffer", stock_concentration: "5 x", final_concentration: "1 x", volume_reaction: "5.0 ul")
prot_reag14 = ProtocolReagent.create(protocol_id: "2", name: "dNTPs", stock_concentration: "2.5 mM", final_concentration: "250 uM", volume_reaction: "2.5 ul")
prot_reag15 = ProtocolReagent.create(protocol_id: "2", name: "Diluted Herculase DNA Pol", stock_concentration: "-", final_concentration: "-", volume_reaction: "0.25 ul")
prot_reag16 = ProtocolReagent.create(protocol_id: "2", name: "Total", stock_concentration: "-", final_concentration: "-", volume_reaction: "22.5 ul")
prot_reag17 = ProtocolReagent.create(protocol_id: "3", name: "H2O", stock_concentration: "-", final_concentration: "-", volume_reaction: "12.75 ul")
prot_reag18 = ProtocolReagent.create(protocol_id: "3", name: "Herculase Bffer", stock_concentration: "5 x", final_concentration: "1 x", volume_reaction: "5.0 ul")
prot_reag19 = ProtocolReagent.create(protocol_id: "3", name: "dNTPs", stock_concentration: "2.5 mM", final_concentration: "250 uM", volume_reaction: "2.5 ul")
prot_reag20 = ProtocolReagent.create(protocol_id: "3", name: "Diluted Herculase DNA Pol", stock_concentration: "-", final_concentration: "-", volume_reaction: "0.25 ul")
prot_reag21 = ProtocolReagent.create(protocol_id: "3", name: "Total", stock_concentration: "-", final_concentration: "-", volume_reaction: "20.5 ul")
prot_reag22 = ProtocolReagent.create(protocol_id: "9", name: "H2O", stock_concentration: "-", final_concentration: "-", volume_reaction: "4.25 ul")
prot_reag23 = ProtocolReagent.create(protocol_id: "9", name: "GoTaq Green", stock_concentration: "2 x", final_concentration: "1 x", volume_reaction: "6.25 ul")
prot_reag24 = ProtocolReagent.create(protocol_id: "9", name: "Forward Primer", stock_concentration: "-", final_concentration: "-", volume_reaction: "0.5 ul")
prot_reag25 = ProtocolReagent.create(protocol_id: "9", name: "Reverse Primer", stock_concentration: "-", final_concentration: "-", volume_reaction: "0.5 ul")
prot_reag26 = ProtocolReagent.create(protocol_id: "9", name: "Total", stock_concentration: "-", final_concentration: "-", volume_reaction: "11.5 ul")
prot_reag27 = ProtocolReagent.create(protocol_id: "5", name: "Cloning Buffer", stock_concentration: "-", final_concentration: "-", volume_reaction: "4.5 ul")
prot_reag28 = ProtocolReagent.create(protocol_id: "5", name: "Vector Mix", stock_concentration: "-", final_concentration: "-", volume_reaction: "1.5 ul")

prot1 = Protocol.create(process: "Spcr", name: "cDNA PCR", content: "<br><br><ul><li>Please be aware that these steps are automated by the Bioparts database for you.  Think about these steps and ensure they are appropriate for your experiment.  A database will not always be available to tell you what to do and how to do it.</li><li>Controls must be used with every set of PCR reactions.</li></ul><br><br><ol><li>Sign up for a PCR machine.</li><li>Thaw reagents on ice; keep Herculase on ice AT ALL TIMES!</li><li>When each reagent has thawed, vortex briefly before pipetting.</li><li>Prepare Master Mix (MM) for all PCRs (as shown below). Be sure to add Herculase last.</li><li>Vortex MM briefly and aliquot 21 micL to each PCR tube.</li><li>Add 2.0 micL of template DNA to PCR reaction tubes.</li><li>With exception of control PCR tubes, add 2.0 micL of primer mix to each PCR tube (forward and reverse primers are mixed together for a 3.0 micM concentration of each) </li><li>Add 2.0 micL of positive control and negative control to control PCR tubes</li><li>Run reactions in PCR machine with following program:</li></ol><br><br><b>Standard PCR Program</b><br><br>  94&deg;C, 3 minutes<br><br><b>  30 cycles:</b><br>  94&deg;C, 30 seconds<br>  55&deg;C, 30 seconds<br>  68&deg;C, 30 seconds<br><br>  68&deg;C, 3 minutes<br>  4&deg;C, hold<br>", scaling_factor: 3, user: user)
prot2 = Protocol.create(process: "Tpcr", name: "tPCR", content: "<b>Templeteless PCR Protocol</b><br><br><ul><li>Please be aware that these steps are automated by the Bioparts database for you.  Think about these steps and ensure they are appropriate for your experiment.  A database will not always be available to tell you what to do and how to do it.</li><li>Positive control must be used with every set of PCR reactions.</li></ul><br><br><ol><li>Sign up for a PCR machine.</li><li>Thaw reagents on ice; keep Herculase on ice AT ALL TIMES!</li><li>When each reagent has thawed, vortex briefly before pipetting.</li><li>Prepare Master Mix (MM) for all PCRs (as shown below). Be sure to add Herculase last.</li><li>Vortex MM briefly and aliquot 22.5 micL to each PCR tube.</li><li>Add 2.5 micL of TPM to each PCR reaction tubes and 2.5 micL of positive control to control PCR tube.</li><li>Run reactions in PCR machine with following program:</li></ol><br><br><b>Templeteless PCR Program (touchdown)</b><br><br><li>  94&deg;C, 3 minutes<br><br><b>  5 cycles:</b><br>  94&deg;C, 30 seconds<br>  69&deg;C, 30 seconds<br>  72&deg;C, 15 seconds<br><br><b>  5 cycles:</b><br>  94&deg;C, 30 seconds<br>  65&deg;C, 30 seconds<br>  72&deg;C, 15 seconds<br><br><b>  5 cycles:</b><br>  94&deg;C, 30 seconds<br>  61&deg;C, 30 seconds<br>  72&deg;C, 15 seconds<br><br>  72&deg;C, 3 minutes<br>  10&deg;C, hold<br>", scaling_factor: 3, user: user)
prot3 = Protocol.create(process: "Fpcr", name: "fCPR", content: "<b>Finish PCR Protocol</b><br><br><ul><li>Please be aware that these steps are automated by the Bioparts database for you.  Think about these steps and ensure they are appropriate for your experiment.  A database will not always be available to tell you what to do and how to do it.</li><li>Controls must be used with every set of PCR reactions.</li></ul><br><br><ol><li>Sign up for a PCR machine.</li><li>Dilute tPCR products 1:5 in H2O.  To do so, add 5 micL of tPCR product to 20 micL of H2O in a 1.5 mL Eppendorf tube.  Please note that the overall dilution factor is 1:50 as you will ultimately put only 2.5 micL of this dilution into a 25 micL reaction.  The templateless PCR product concentration is difficult to determine due to its herogeneity.  This dilution factor is something that you can vary in order to troubleshoot the success of the fPCR.</li><li>Thaw reagents on ice; keep Herculase on ice AT ALL TIMES!</li><li>When each reagent has thawed, vortex briefly before pipetting.</li><li>Prepare Master Mix (MM) for all PCRs (as shown below). Be sure to add Herculase last.</li>
<li>Vortex MM briefly and aliquot 20.5 micL of the reaction cocktail to each PCR tube.</li><li>Add 2.0 micL of OPM (3 micM outer primer mix) to the fPCR reaction tubes and 2.0 micL of H2O to the positive control reaction tube.</li><li>Add 2.0 micL of diluted tPCR product to fPCR reaction tubes and 2.5 micL of positive control to the control reaction tube.</li><li>Run reactions in PCR machine with following program:</li></ol><br><br><b>Finish PCR Program</b><br><br>  94&deg;C, 3 minutes<br>  55&deg;C, 30 seconds<br>  72&deg;C, 15 seconds<br><br><b>  25 cycles:</b><br>
  94&deg;C, 30 seconds<br>  55&deg;C, 30 seconds<br>  72&deg;C, 15 seconds<br><br>  72&deg;C, 3 minutes<br>  10&deg;C, hold<br>", scaling_factor: 3, user: user)
prot4 = Protocol.create(process: "PcrGel", name: "PCR Gel", content: "<b>Herculase</b><br><br><ul>
<li>Wear gloves throughout the entire gel making process.</li><li>Clean up after yourself.  Rinse all glass or plastic wear after each use and let dry over the sink.</li></ul><br><br><li><u>Preparing your Small Gel (100 mL, 1% agarose/TTE)</u></li><br><br><ol><li>Prepare 500 mL 1x TTE by adding 25 ml 20x TTE buffer and 470 ml distilled water (white faucet by the door) into the dedicated 2 L graduated cylinder.</li><li>Double parafilm top make sure it does not leak.  Remaining buffer will be used for your gel.  Reusing old buffer is not recommended as its concentration changes with evaporation.</li><li>Measure out 1.0 g of Agarose (keep balance area cleaner than you found it).</li><li>Put 1.0 g Agarose into 500 mL bottle and add 100 mL 1x TTE.  Swirl thoroughly</li><li>Microwave for 1 minute or until boiling starts.</li><li>Remove from microwave with great caution and swirl ONLY once it stops bubbling completely.</li><li>Return to microwave for 15 sec.</li><li>Remove and again swirl ONLY ONCE until it stops bubbling completely.</li><li>Repeat previous two steps as needed until agarose solution is completely clear and there is no partially molten agarose on the bottom of the bottle.</li><li>Place in 60 degree oven for at least 30 minutes. Or leave at room temperature for 5 minutes, then swirl under tap water wearing an autoclave glove, until the bottle is cool enough not to feel burning hot to bare hand.</li><li>Wash and dry your gel tray thoroughly; seal with tape with &quot;hospital&quot; corners. You cannot be too careful about getting a good seal or you will need to do it all over again. You can also use the rubber seals but if the gel tray is damaged you may find this frustrating. Wash and dry comb(s).</li><li>WEAR GLOVES!!! Add 12.5 mL of 10mg/mL Ethidium Bromide. Swirl, pour your agarose gel into gel tray. Wait until gel is solidified (~30 min) before removing comb by pulling gently upward.</li><li>Pour remaining 1x buffer into gel box and lower gel into buffer. Load gel (do not forget PCR controls and MW standard!) and run at 100 V.</li>
</ol><br><br><li><u>Running your gel</u></li><br><br><ol><li>Make a diluted loading dye master mix by combining 4.0 micL of H2O for every 1.0 micL of loading dye (multiply by the number of PCR reactions plus ~2).</li><li>On a piece of parafilm, combine 5.0 micL of each PCR product (do not forget the controls!) with 5.0 micL of diluted loading dye master mix.  Do not allow samples to sit on parafilm for too long, the liquid beads will evaporate.</li><li>Load all 10.0 micL of each PCR sample into agarose wells.  Save a spot for the ladder.</li><li>Load 6.0 micL of 2-log ladder.</li><li>Place gel lid with electrodes on gel box.  Remember that DNA runs from negative (black) to positive (red).  Check your leads!</li><li>Run the gel at 100V.  Confirm that there is resistance in the circuit by checking the Amps.  Zero amps means the leads are not properly connected.  If the amps measures off the scale, consider replacing the buffer.  Also check for bubbles rising from both electrodes.</li><li>Run gel approximately 20 to 30 minutes, then check dye front.  Allow the gel to continue running as necessary.</li></ol><br><br><li><u>Photography on the Kodak GelLogic Gel Imager</u></li><br><br><ol><li>Place gel inside imager.</li><li>Click on Kodak 1D icon and select &quot;Capture GL100&quot; and then &quot;Preview&quot; to view your gel.</li>
<li>Turn on Epi White (and set timer switch to 10).</li><li>Adjust gel position, zoom (top ring on camera) so edges of gel are visible, and focus (middle ring) till sharp.</li><li>Turn switch to UV transillumination and select &quot;Capture Image&quot;.</li><li>Select &quot;File&quot; and then &quot;Export Image&quot; to save your image.  Make sure you save to a .jpg format.</li><li>Upload your image to the Bioparts database.  You may want to label the lanes in powerpoint or photoshop. You can load up both annotated and unannotated image but make sure you always upload the raw gel image.</li></ol><br>", scaling_factor: 3, user: user)
prot5 = Protocol.create(process: "Ligation", name: "Ligation", content: "<br><br><li><u>Using StrataClone Blunt PCR Coning Kit</u></li><br><br><ol><li>Dilute successful F-PCR products 1:10 (1ul of F-PCR in 9ul of H2O).</li><li>Make ligation cocktail as shown below and mix gentle pipetting.</li><li>Aliquot 2ul of the cocktail per reaction tube. Add 1ul of your diluted F-PCR product and mix by gently pipetting.</li>
<li>Incubate reaction at room temperature for 5 min. Place reaction on ice after 5 min.  If you cannot continue on to the Transformation step, store reactions at -20&deg;C for later use.</li></ol><br>", scaling_factor: 3, user: user)
prot6 = Protocol.create(process: "Spcr", name: "Genomic DNA PCR", content: "<b>Genomic PCR Protocol</b><br><br><ul><li>Please be aware that these steps are automated by the Bioparts database for you.  Think about these steps and ensure they are appropriate for your experiment.  A database will not always be available to tell you what to do and how to do it.</li><li>Controls must be used with every set of PCR reactions.</li></ul><br><br><ol><li>Sign up for a PCR machine.</li><li>Thaw reagents on ice; keep Herculase on ice AT ALL TIMES!</li><li>When each reagent has thawed, vortex briefly before pipetting.</li><li>Prepare Master Mix (MM) for all PCRs (as shown below). Be sure to add Herculase last.</li><li>Vortex MM briefly and aliquot 23 micL to each PCR tube.</li><li>With exception of control PCR tubes, add 2.0 micL of primer mix to each PCR tube (forward and reverse primers are mixed together for a 3.0 micM concentration of each) </li><li>Add 2.0 micL of positive control and negative control to control PCR tubes</li><li>Run reactions in PCR machine with following program:</li></ol><br><br><b>Standard PCR Program</b><br><br>  94&deg;C, 3 minutes<br><br><b>  30 cycles:</b><br>  94&deg;C, 30 seconds<br>  55&deg;C, 30 seconds<br>  68&deg;C, 30 seconds<br><br>  68&deg;C, 3 minutes<br>  4&deg;C, hold<br>", scaling_factor: 3, user: user)
prot7 = Protocol.create(process: "Transformation", name: "Transformation", content: "<br><br><li><u>Preparation:</u></li><br><br><ul><li>This is the easiest protocol to &quot;mess up&quot; in the entire workflow.  Cells must be kept cold at all times!</li><li>Chill labeled tubes, ligation mixes, etc., on ice prior to mixing of cells and DNA (ligation mix).</li><li>It is crucial to use only the StrataClone SoloPack cells because they express the necessary Cre recombinase.  Using other competent cells will not work!</li><li>One tube of competent cells is to be used for 2 transformation reactions.  If you have an odd number of reactions try to coordinate with someone else who has an odd number as well.</li><li>Never thaw frozen cells in your hand; always plunge frozen cells in ice and allow to thaw there.  As soon as they thaw, they start dying.  The more time passes (up to the point where SOC is added) the more they die and the less transformants you&#39;ll get.  Also, never heat the cells prematurely; make sure your timing for the heat shock step is precise.</li><li>Check to make sure the water bath is holding at 42 o C.  If it is not, but is just slightly under(still above 40 o C), that&#39;s fine.  If it is too hot add a small amount of ice to bring it down; notify TAs so temperature can be adjusted.  Do not adjust the temperature yourself.</li><li>Whenever you are incubating or storing plates, they should be stored upside down to prevent condensation from dripping onto your plates.</li></ul><br><br><li><u>Reaction Conditions:</u></li><br><br><ol><li>Get one LB/Carb/IPTG/Xgal plate per transformation from 4&deg;C, label, and put on bench top.</li><li>Obtain LB medium and make sure it&#39;s sterile (it should be totally transparent), and at room temperature. The TAs should have this prepared for you in advance.</li><li>Check to make sure the water bath is holding at 42&deg;C.</li><li>Add 3 micL H2O to each ligation reaction and mix well by gently pipetting.</li><li>Thaw one tube of StrataClone SoloPack competent cells per 2 ligation reactions on ice. If you have an odd number of reactions try to coordinate with someone else who has an odd number as well.</li>
<li>Move 25 micL of thawed competent cells into a new, pre-cooled 1.5 mL Eppendorf tube.  Do NOT use the original competent cell tube for the transformation since it does not have a tapered bottom and will not fit snugly into the heat block (leading to non-uniform temperature shock).</li>
<li>Add 1.0 micL of the diluted ligation reaction to the competent cells. Mix gently by &quot;finger flicking&quot;, do not pipet up and down.</li>
<li>Incubate the reaction on ice for 20 minutes.</li><li>Heat shock the transformation reaction at 42&deg;C for 45 seconds no more, no less.</li><li>Immediately place the transformation reaction back on ice for 2 minutes.</li><li>Add 125 micL room temperature LB medium to the transformation reaction. Put cells in the 30&deg;C incubator for 1 hr, or in the 37&deg;C incubator for 30 min.</li><li>Resuspend cells by gentle pipetting. Plate 100 micL of the transformation mix on LB +carb +X-gal plates (shake with 4-6 glass beads).  Save the other 50 on your benchtop &ndash; if you have not so many colonies the next day, you can plate the other 50 and get more.</li><li>Dump out glass beads into trash.</li><li>Incubate plates overnight at 30&deg;C (ensure plates are upside-down).  Plates should be incubated for around 24-36 hours. If you are not ready to move to the next step, you may hold them at 4&deg;C wrapped in parafilm.</li></ol><br>", scaling_factor: 3, user: user)
prot8 = Protocol.create(process: "Cloning", name: "Cloning", content: "<br><br><ol><li>White colonies are picked individually into 150 micL LB/Carb in sterile 96 well plates (with a lid) and grown, standing, overnight at 37&deg;C for 18 &ndash; 24h.</li><li>If you have trouble with growth, you can try growing them at 30&deg;C for 24 hours.</li><li>Pick at least 24 clones per BB, or all clones that you have if you have less, so you will be able to submit at least 12-18 to sequencing after checking for correct size product after CS-PCR.</li><li>For each clone, filling up the 96-well culture plate row by row (e.g. first BB going from A1 to B12) might help if you want to use the multichannel pipet for loading gels.
The culture is used as template for PCR the next day (or may be held at 4&deg;C prior to use).</li></ol><br><br>", scaling_factor: 3, user: user)
prot9 = Protocol.create(process: "Cspcr", name: "csPCR", content: "<br><li>Preparation &ndash; the day before</li><br><br><ol><li>White colonies are picked individually into 150 micL LB/Carb in sterile 96 well plates (with a lid) and grown, standing, overnight at 37&deg;C for 18 &ndash; 24h.</li><li>If you have trouble with growth, you can try growing them at 30&deg;C for 24 hours.</li><li>Pick at least 24 clones per BB, or all clones that you have if you have less, so you will be able to submit at least 12-18 to sequencing after checking for correct size product after CS-PCR.</li><li>For each clone, filling up the 96-well culture plate row by row (e.g. first BB going from A1 to B12) might help if you want to use the multichannel pipet for loading gels.  The culture is used as template for PCR the next day (or may be held at 4&deg;C prior to use).</li></ol><br><br><li>Reaction Setup</li><br><br><ul><li>Please be aware that these steps are automated by the Bioparts database for you.  Think about these steps and ensure they are appropriate for your experiment.  A database will not always be available to tell you what to do and how to do it.</li><li>Thaw reagents on ice; keep Herculase on ice AT ALL TIMES!</li><li>When each reagent has thawed, vortex briefly before pipetting.</li><li>Controls must be used with every set of PCR reactions.</li><li>Pipet GoTaq Green slowly to prevent reagents sticking to inside of pipet.</li><li>The appropriate extension time is 1 minute per 1kb of expected product size. Do not forget that you are adding on about 200bp to your PCR product due to flanking regions of the vector that are between the forward and reverse primer sites.</li></ul><br><br><ol><li>Sign up for a PCR machine.</li><li>Prepare Master Mix (MM) for all PCRs (as shown below).</li><li>Vortex MM briefly and aliquot 11.5? micL of the reaction cocktail to each well on a 96-well plate.</li><li>Add 1.0 micL of bacterial cultures to csPCR reaction wells and 1.0 micL of positive control to control well (Be sure to mix individual cultures thoroughly prior to pipetting.  If you do not, you may transfer too many or too few cells and the PCR will not work).</li><li>Run reactions in PCR machine with following program:</li></ol><br><br><b>Colony Screening PCR Program</b><br><br>  94&deg;C, 4 minutes<br><br><b>  30 cycles:</b><br>  94&deg;C, 30 seconds<br>  55&deg;C, 30 seconds<br>  72&deg;C, 1 minute<br><br>  72&deg;C, 3 minutes<br>  10&deg;C, hold<br>", scaling_factor: 3, user: user)
prot10 = Protocol.create(process: "CspcrGel", name: "csPCR Gel", content: "<b>GoTaq Green</b><br><br><ul><li>Wear gloves throughout the entire gel making process.</li><li>Clean up after yourself.  Rinse all glass or plastic wear after each use and let dry over the sink.</li></ul><br><br><li><u>Preparing your Large Gel (l250 mL, 1% agarose/TTE)</u></li><br><br><ol><li>Prepare 2000 mL 1x TTE by adding 100 ml 20x TTE buffer and 1900 ml distilled water (white faucet by the door) into the dedicated 2 L graduated cylinder.</li><li>Double parafilm top make sure it does not leak.  Remaining buffer will be used for your gel.  Reusing old buffer is not recommended as its concentration changes with evaporation.</li><li>Measure out 2.5 g of Agarose (keep balance area cleaner than you found it).</li><li>Put 2.5 g Agarose into 500 mL bottle and add 100 mL 1x TTE.  Swirl thoroughly</li><li>Microwave for 2.5 minute or until boiling starts.</li><li>Remove from microwave with great caution and swirl ONLY once it stops bubbling completely.</li><li>Return to microwave for 15 sec.</li><li>Remove and again swirl ONLY once until it stops bubbling completely.</li><li>Repeat previous two steps as needed until agarose solution is completely clear and there is no partially molten agarose on the bottom of the bottle.</li><li>Place in 60 degree oven for at least 30 minutes. Or leave at room temperature for 5 minutes, then swirl under tap water wearing an autoclave glove, until the bottle is cool enough not to feel burning hot to bare hand.</li><li>Wash and dry your gel tray thoroughly; seal with tape with &quot;hospital corners.&quot; You cannot be too careful about getting a good seal or you will need to do it all over again. You can also use the rubber seals but if the gel tray is damaged you may find this frustrating. Wash and dry comb(s).</li><li>WEAR GLOVES!!! Add 12.5 mL of 10mg/mL Ethidium Bromide. Swirl, pour your agarose gel into gel tray. Wait until gel is solidified (~30 min) before removing comb by pulling gently upward.</li><li>Pour remaining 1x buffer into gel box and lower gel into buffer.</li></ol><br><br><li><u>Running your gel</u></li><br><br><ol><li>Load 5.0 micL of each PCR sample (GoTaq Green) into agarose wells.  Do not forget the controls!  The csPCR control only needs 1-2 micL loaded into the well.  Save a spot for the ladders.</li><li>Load 6.0 micL of 2-log ladder.</li><li>Place gel lid with electrodes on gel box.  Remember that DNA runs from negative (black) to positive (red).  Check your leads!</li><li>Run the gel at 125V.  Confirm that there is resistance in the circuit by checking the Amps.  Zero amps means the leads are not properly connected.  If the amps measures off the scale, consider replacing the buffer.  Also check for bubbles rising from both electrodes.</li><li>Run gel approximately 20 to 30 minutes, then check dye front.  Allow the gel to continue running as necessary.</li></ol><br><br><li><u>Photography on the Kodak GelLogic Gel Imager</u></li><br><br><ol><li>Place gel inside imager.</li><li>Click on Kodak 1D icon and select &quot;Capture GL100&quot; and then &quot;Preview&quot; to view your gel.</li><li>Turn on Epi White (and set timer switch to 10).</li><li>Adjust gel position, zoom (top ring on camera) so edges of gel are visible, and focus (middle ring) till sharp.</li><li>Turn switch to UV transillumination and select &quot;Capture Image&quot;.</li><li>Select &quot;File&quot; and then &quot;Export Image&quot; to save your image.  Make sure you save to a .jpg format.</li><li>Upload your image to the Bioparts database.  You may want to label the lanes in powerpoint or photoshop. You can load up both annotated and unannotated image but make sure you always upload the raw gel image.</li></ol><br>", scaling_factor: 3, user: user)
prot11 = Protocol.create(process: "Sequencing", name: "Submitting Clones for Sequencing", content: "<br><br><li>**Please read this all the way through before starting the sequence submission</li><br><br><ol><li>If you did your csPCR and have clones with an insert of the right size (keep in the mind that you are looking for building blocks of the right size.  E.g. 750bp building block plus ~200bp extra flanking sequences from the vector), you can send them for sequencing. Please submit up to 8 clones for each building block, if you have them.</li><li>In the Bioparts database, click on the &quot;Submit clones for sequencing&quot; link. Choose the clones that you want to send.</li><li>Biopartsdb will assign your clones to certain wells in a 96-well plate.  Most likely, other wells on the plate map will be blocked out with red circles &ndash; this means that someone else has signed up for those wells.</li><li>Get a new 96-well plate and label it with your name and the plate number indicated by Biopartsdb (like &quot;1005&quot;).  Even if you are continuing a plate that has already been started (i.e. there are red circles on the plate map), YOU DO NOT NEED TO LOCATE THAT PLATE. You will use your own plate (see below).</li><li>You will be transferring the rest of the bacterial cultures that you used for the csPCR reactions into the assigned wells.</li><li>For example, if the plate map shows red circles in wells A1-B12, and Biopartsdb assigned you wells C1-H12, please put your cultures in wells C1-H12. The rest of this plate (Rows A and B) will be empty.  Someone else will be submitting a plate that has cultures in Rows A and B, and the rest of their plate will be empty. At the end, I will consolidate everything.</li><li>Bottom line: Put your cultures into your own plate and fill only those wells that you signed up for. The rest of the wells (i.e. those indicated by the red circles on the the Biopartsdb plate map) will be empty.</li><li>Transfer the rest of the bacterial cultures that you used for the csPCR reactions into the wells assigned by the Biopartsdb.  BE SURE TO PIPET THE CULTURES UP AND DOWN before transferring them since the bacteria will have settled to the bottom of the plate while sitting in the fridge.</li><li>If it has been a while since you grew them up (more than two weeks), you should regrow them as follows. Set up a 96-well plate with 150 micL LB-carb in each relevant well as specified by Biopartsdb.</li><li>After pipetting up and down to mix, transfer 1 micL from your source plate to this plate. Grow up at 30 C for 24 hours. Check to make sure bacteria grew (wells should be turbid.  If not check with an instructor or TA. Submit this plate, labeled as above.</li><li>Carefully seal the plate with a foil seal. Please label the seal with your name and plate number as well!</li><li>Put the plate in the silver fridge in the drawer marked &quot;Clones for sequencing drop-off.&quot;</li></ol><br>", scaling_factor: 3, user: user)
